Test/Platform/Tests/CLR/mscorlib/systemlib/systemlib2/Utf8EncodingTests.cs

```diff
@@ -76,5 +76,24 @@ public MFTestResults Utf8EncodingTests_Test2()
 
             return result ? MFTestResults.Pass : MFTestResults.Fail;
         }
+
+        [TestMethod]
+        public MFTestResults Utf8EncodingTests_Test3()
+        {
+            // This tests involves a string with a special character
+
+            bool result = true;
+
+            string str = "AB\u010DAB";
+
+            byte[] data = new byte[4];
+            int count = UTF8Encoding.UTF8.GetBytes(str, 1, 3, data, 0);
+
+            result &= count == 4;
+
+            result &= (new string(UTF8Encoding.UTF8.GetChars(data)) == "B\u010DA");

```

Comment:
We should generally use logical && for this type of operation, rather than bitwise &.  We should also probably short-circuit when anything fails, so something like the following (which mirrors the chained "if (SUCCEEDED(hr))"  pattern):

if (result)
{
    result = expr;
}

if (result)
{
    ...
}